src/socketio.quiver.js

```diff
@@ -970,23 +1140,107 @@ QuiverSocialProvider.prototype.changeRoster = function(userId, clientSuffix, inv
 };
 
 /**
- * Interpret messages from the server to this as owner.
- * There are 3 types of messages
- * - Directed messages from endpoints (message)
- * - State information from the server on initialization (state)
- * - Roster change events (users go online/offline) (roster)
+ * @param {!QuiverSocialProvider.server_} server The server through which the
+ *     message was delivered.
+ * @param {Object} msg Encrypted message from the server
+ * @private
+ **/
+QuiverSocialProvider.prototype.onEncryptedMessage_ = function(server, msg) {
+  if (!msg) {
+    return;
+  }
+
+  this.pgp_.verifyDecrypt(msg.cipherText, msg.key).then(function(plain) {
```

Comment:
Is it still the case that pings are unencrypted (but now signed)?  If so does this mean that msg.key would be set to our friends publicKey to verify that it's signed, but msg.cipherText would actually be plain text?